_build/test/Tests/Model/modParserTest.php

```diff
@@ -646,7 +649,172 @@ public function providerProcessElementTags() {
+
+        $c = $content;
+
+        $this->modx->parser->processElementTags(
+            $params['parentTag'],
+            $content,
+            $params['processUncacheable'],
+            $params['removeUnprocessed'],
+            $params['prefix'],
+            $params['suffix'],
+            $params['tokens'],
+            $params['depth']
+        );
+
+        $set->remove();
+        $snippet->remove();
+        $join->remove();
+
+        $this->assertEquals($expected, $content, "Did not get expected results from parsing {$c}.");
+    }
+
+    public function providerPropertySetCall()
+    {
+        // In this test, snippetName and propSetName are replaced with a random string
+        // for each run
+        return [
+            [
+                '[[snippetName? &prop=`123`]]',
+                '123',
+                [],
+                [
+                    'parentTag' => '',
+                    'processUncacheable' => true,
+                    'removeUnprocessed' => false,
+                    'prefix' => '[[',
+                    'suffix' => ']]',
+                    'tokens' => [],
+                    'depth' => 10
+                ]
+            ],
+            [
+                '[[snippetName@propSetName]]',
+                '123',
+                [
+                    'prop' => '123',
+                ],
+                [
+                    'parentTag' => '',
+                    'processUncacheable' => true,
+                    'removeUnprocessed' => false,
+                    'prefix' => '[[',
+                    'suffix' => ']]',
+                    'tokens' => [],
+                    'depth' => 10
+                ]
+            ],
+            [
+                '[[snippetName@propSetName? &otherProp=`foo`]]',
+                '789',
+                [
+                    'prop' => '789',
+                ],
+                [
+                    'parentTag' => '',
+                    'processUncacheable' => true,
+                    'removeUnprocessed' => false,
+                    'prefix' => '[[',
+                    'suffix' => ']]',
+                    'tokens' => [],
+                    'depth' => 10
+                ]
+            ],
+            [
+                '[[snippetName@propSetName? &prop=`123`]]',
+                '123',
+                [
+                    'prop' => '456', // needs to be ignored because &prop is specified as override
+                ],
+                [
+                    'parentTag' => '',
+                    'processUncacheable' => true,
+                    'removeUnprocessed' => false,
+                    'prefix' => '[[',
+                    'suffix' => ']]',
+                    'tokens' => [],
+                    'depth' => 10
+                ]
+            ],
+            [
+                'This is a [[snippetName@propSetName:default=`default value`]]',
+                'This is a default value',
```

Comment:
This is the test that's failing. The snippet returns the value of the `prop` property, which doesn't exist in this scenario, so the snippet returns an empty value. That should make it return the `default` filter, but instead it returns nothing at all (`This is a `). 

Can reproduce the same behavior when creating things manually on a template/resource, so that suggests _something_ is broken?

Reply:
There is a chance this may've been related to #16337, so will need to test again.